src/java/main/org/apache/zookeeper/common/ZKConfig.java

```diff
@@ -0,0 +1,222 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zookeeper.common;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+
+import org.apache.zookeeper.Environment;
+import org.apache.zookeeper.server.quorum.QuorumPeerConfig.ConfigException;
+import org.apache.zookeeper.server.util.VerifyingFileFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is used as base class for ZooKeeper Client and Server
+ * configuration. Its support to read client configuration from system property
+ * and configuration file. User can override system property by
+ * {@link #setProperty(String, String)} API.
+ */
+
+public class ZKConfig {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ZKConfig.class);
+    public static final String SSL_KEYSTORE_LOCATION = "zookeeper.ssl.keyStore.location";
+    public static final String SSL_KEYSTORE_PASSWD = "zookeeper.ssl.keyStore.password";
+    public static final String SSL_TRUSTSTORE_LOCATION = "zookeeper.ssl.trustStore.location";
+    public static final String SSL_TRUSTSTORE_PASSWD = "zookeeper.ssl.trustStore.password";
+    public static final String SSL_AUTHPROVIDER = "zookeeper.ssl.authProvider";
+    public static final String JUTE_MAXBUFFER = "jute.maxbuffer";
+    public static final String KINIT_COMMAND = "zookeeper.kinit";
+    public static final String JGSS_NATIVE = "sun.security.jgss.native";
+
+    private Map<String, String> properties = new HashMap<String, String>();
+
+    /**
+     * values are ini
+     */
+    public ZKConfig() {
+        init();
+    }
+
+    /**
+     *
+     * @param configPath
+     *            Configuration file path
+     * @throws ConfigException
+     *             if failed to load configuration properties
+     */
+
+    public ZKConfig(String configPath) throws ConfigException {
+        this(new File(configPath));
+    }
+
+    /**
+     *
+     * @param configFile
+     *            Configuration file
+     * @throws ConfigException
+     *             if failed to load configuration properties
+     */
+    public ZKConfig(File configFile) throws ConfigException {
+        init();
+        addConfiguration(configFile);
+    }
+
+    private void init() {
+        /**
+         * backward compatibility for all currently available client properties
+         */
+        handleBackwardCompatibility();
```

Comment:
In which case do we need to handle backward compatibility? Accessing these properties is internal to ZK, so I'm wondering why we need to worry about it at all.